src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -0,0 +1,356 @@
+                    Collections.singletonMap(Parameter.LEVEL, "program"));
+
+    public static final Param NAMESPACE =
+            new Param("namespace", URI.class, "Namespace prefix", false);
+    public static final Param DATASTORE_NAME =
+            new Param("datastorename", String.class, "Name of the datastore", true);
+    public static final Param ROOT_ENTITY =
+            new Param("root entity", String.class, "Root Entity", true);
+    public static final Param DATASTORE_METADATA =
+            new Param("datastore", String.class, "JDBC related DataStore", true);
+    public static final Param DOMAIN_MODEL_EXCLUSIONS =
+            new Param(
+                    "excluded objects",
+                    String.class,
+                    "Excluded comma separated domainmodel object list",
+                    false);
+
+    @Override
+    public String getDisplayName() {
+        return "Smart Data Loader";
+    }
+
+    @Override
+    public String getDescription() {
+        return "AppSchema smart builder tool";
+    }
+
+    @Override
+    public final Param[] getParametersInfo() {
+        LinkedHashMap<String, DataAccessFactory.Param> map = new LinkedHashMap<>();
+        setupParameters(map);
+        return map.values().toArray(new Param[map.size()]);
+    }
+
+    @Override
+    public boolean isAvailable() {
+        try {
+            Class.forName(getDriverClassName());
+            return true;
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean canProcess(Map<String, ?> params) {
+        try {
+            Object dbType = DBTYPE.lookUp(params);
+            return DBTYPE_STRING.equals(dbType)
+                    && DataUtilities.canProcess(params, getParametersInfo());
+        } catch (Exception e) {
+            // do nothing. based on AppSchemaDataAccessFactory code
+        }
+        return false;
+    }
+
+    public String getDriverClassName() {
+        return "org.postgresql.Driver";
+    }
+
+    private void setupParameters(Map<String, Param> parameters) {
+        parameters.put(DBTYPE.key, DBTYPE);
+        parameters.put(NAMESPACE.key, NAMESPACE);
+        parameters.put(DATASTORE_NAME.key, DATASTORE_NAME);
+        parameters.put(DATASTORE_METADATA.key, DATASTORE_METADATA);
+        parameters.put(ROOT_ENTITY.key, ROOT_ENTITY);
+        parameters.put(DOMAIN_MODEL_EXCLUSIONS.key, DOMAIN_MODEL_EXCLUSIONS);
+    }
+
+    private String getFilenamePrefix(Map<String, Serializable> params) throws IOException {
+        String rootEntity = lookup(ROOT_ENTITY, params, String.class);
+        String filenamePrefix = rootEntity;
+        return filenamePrefix;
+    }
+    /**
+     * Helper method that based on parameters, builds domainmodel, generates associated mapping
+     * files and saves them in the workspace, returning the resulting DataStore.
+     */
+    private DataAccess<FeatureType, Feature> createDataStore(
```

Comment:
Looks like this method does several things and is getting big, maybe place for split it into smaller methods with single tasks?

Reply:
fixed